app/models/site_setting.rb

```diff
@@ -121,6 +121,11 @@ def self.attachment_filename_blacklist_regex
     @attachment_filename_blacklist_regex ||= Regexp.union(SiteSetting.attachment_filename_blacklist.split("|"))
   end
 
+  def self.unicode_username_character_whitelist_regex
+    @unicode_username_whitelist_regex = SiteSetting.unicode_username_character_whitelist.present? \
```

Comment:
I wanted to memoize the regex, but that broke my specs. Looks like saving a site setting does not remove memoized values. I guess `SiteSetting.attachment_content_type_blacklist_regex` and `SiteSetting.attachment_filename_blacklist_regex` will have the same problem.